crypto/evp/p_lib.c

```diff
@@ -493,7 +493,7 @@ int EVP_PKEY_set1_RSA(EVP_PKEY *pkey, RSA *key)
 
 RSA *EVP_PKEY_get0_RSA(const EVP_PKEY *pkey)
 {
-    if (pkey->type != EVP_PKEY_RSA) {
+    if (pkey->type != EVP_PKEY_RSA && pkey->type != EVP_PKEY_RSA_PSS) {
```

Comment:
how did you create this object?
with EVP_PKEY_assign_RSA?

Reply:
I can't say for sure, but most likely using one of `PEM_read_bio_PrivateKey`, `d2i_PKCS8PrivateKey_bio`, `EVP_PKCS82PKEY`, `d2i_PUBKEY`, or `X509_get_pubkey` depending on the input format and context.

Reply:
You know, you are supposed to use EVP_PKEY_sign with the private key,
with the way you want to do it it will be an OAEP signature?

Reply:
I meant: EVP_DigestSign

Reply:
I am not sure what you mean, could you elaborate?  Why do you think this change is related to signatures?

Reply:
You are not supposed to use the RSA key directly,  creating a PSS signature
requires you to know the correct PSS-parameters, and would simply be ugly
to do with EVP_PKEY_get0_RSA, RSA_padding_add_PKCS1_PSS_mgf1 and
RSA_private_encrypt.  If you still want to do it you can of course
go with `(RSA*)EVP_PKEY_get0(evp)`.
The correct way to do it is using EVP_MD_CTX_new, EVP_DigestSignInit, EVP_DigestSignUpdate, and EVP_DigestSignFinal.
If you do it this way it will also work for ECC keys without any changes.

Reply:
Again, I am not trying to use the returned `RSA*` to sign data, I need to find out how large the modulus is and what the public exponent of the key is. (Copied from my last comment.)

Reply:
For number of bits of the public exponend use EVP_PKEY_size,
I don't understand what you could do with the public exponent,
but if you want to export a public key from the private key, you can do that with
i2d_PublicKey in principle.

Reply:
I am trying to implement a crypto API that conforms to the WebCrypto standard, which requires users to be able to retrieve the public exponent.

Reply:
>  If you still want to do it you can of course go with `(RSA*)EVP_PKEY_get0(evp)`.

Thank you for the hint, this seems to be a good workaround for now!

Reply:
Regardless of the above discussion, it seems a strange thing that `EVP_PKEY_get0_RSA` can't return an RSA-PSS key, so this change seems reasonable to me. Probably even for backport to 1.1.1